include/dlaf/common/assert.h

```diff
@@ -114,6 +114,17 @@ inline void do_assert(bool expr, const common::internal::source_location& loc, c
 #define DLAF_ASSERT_HEAVY(Expr, ...)
 #endif
 
+namespace dlaf::internal {
+/// Helper function for silencing unused parameter warning.
+///
+/// This allows to not add std::maybe_unused to each argument of an UNIMPLEMENTED function,
+/// but still keeping the name of arguments, so that in the future the signature will be
+/// ready for implementation.
+template <class... Args>
+void silenceUnusedWarningFor(Args&&...) {}
+
+}
+
```

Comment:
Not sure why the function is defined here with macros. Wouldn't it to be better to define it with the other functions (i.e. after do_assert at line 50)?